### DISCUSSION TITLE
Is there any guide doc how to make a new my own Field?

### DISCUSSION THREAD

**Discussion by KLA6** [orchidsoftware/platform#2094] [Q&A]
As I've roughly checked, to make a new custom Field, these 3 are essential.
- A Field model in app/Orchid/Screens 
- A Blade view in resources/views/platform/fields
- A JS controller in resources/js/cntrollers/fields

BTW, I couldn't understand yet, how to add the field JS to the Mix.

Is there any DOC or EXAMPLE about how to make a custom Field by myself?

Thank you for reading.

**Comment by rowanfuchs** [comment]
it took me a while to find the doc but it's explained here https://orchid.software/en/docs/builder/

**Reply by tabuna** [reply]
I will supplement the answer to see where you can use the JS code here https://orchid.software/en/docs/javascript/
